Katello/smart_proxy_container_gateway
# Unauthenticated podman pulling working -- ignoring all auth

To test:

1) Set up a smart proxy dev box with Pulp 3 

2) Install this smart proxy plugin with this PR
  a) Clone https://github.com/theforeman/smart-proxy
  b) Copy settings from /etc/foreman-proxy/ to the local smart-proxy checkout
  c) Ensure you can bundle install and run bin/smart-proxy
  d) Install the container gateway plugin by adding it to smart-proxy/bundler.d/Gemfile.local.rb
    -> Ex: `gem 'smart_proxy_container_gateway', :path => "/home/vagrant/smart_proxy_container_gateway"`
  e) Copy the plugin's config example to smart-proxy/config/settings.d/container_gateway.yml
    -> You will need to fill out the container_gateway.yml file accordingly
    -> The pulp-client cert and key must be copied from your Katello dev box:
     `/etc/pki/katello/certs/pulp-client.crt and /etc/pki/katello/private/pulp-client.key`
  f) Ensure your docker proxy Apache proxy config looks like the one in the readme
  g) Bundle install, bin/smart-proxy, curl https://`hostname`:9090/features to check that the container_gateway plugin was installed

3) Sync your smart proxy with some container content and try podman pulling it
  - Example: 
```
sudo podman pull centos7-proxy-devel.cannolo.example.com/default_organization-environment-content_view-product-image
```
